File: app/src/main/java/com/sasaj/graphics/drawingapp/data/source/DrawingsRepository.java
package com.sasaj.graphics.drawingapp.data.source;

import com.sasaj.graphics.drawingapp.data.Drawing;

import java.util.List;

/**
 * Created by DS on 4/21/2017.
 */

public interface DrawingsRepository {


    interface LoadDrawingsCallback {

        void onDrawingsLoaded(List<Drawing> drawings);

        void onDataNotAvailable();
    }

    interface LoadDrawingCallback {

        void onDrawingLoaded(Drawing drawings);

        void onDataNotAvailable();
    }

   void  getDrawings(LoadDrawingsCallback callback);

   void  getDrawing(String id, LoadDrawingCallback callback);
}

File: app/src/main/java/com/sasaj/graphics/drawingapp/viewmodel/dependencies/DrawingsRepository.kt
package com.sasaj.graphics.drawingapp.viewmodel.dependencies

import android.graphics.Bitmap

import com.sasaj.graphics.drawingapp.domain.Drawing

/**
 * Created by sjugurdzija on 4/21/2017.
 */

interface DrawingsRepository {

    val drawings: List<Drawing>
    fun saveDrawing(bitmap: Bitmap?)
}
